# Problem 1
<b>Determine</b> the <b>maximum pressure of water vapour</b> in wet hydrogen at 1 atm pressure in which chromium can be heated without oxidation ocurring at 1500K. 
Is the oxidation of Cr by water vapour exothermic or endothermic?

## Solution

Here, we <b>declare needed variables</b>, namely: 
<br>$P_m$ for Pressure max 
<br>T for Temperature 
<br>$\Delta$$G_3$, $\Delta$$G_4$ for the Gibbs Energy of the reactions
<br>$\Delta$$G$ for <br>$P_{COS}$ for Partial Pressure of COS compound
<br>$P_{CO}$ for partial pressure of CO compound

In [136]:
var('Pm',latex_name="\\ P_{max}")
var("T")
var("delG3",latex_name="\\Delta G_3")
var("delG4",latex_name="\\Delta G_4")
var("DG",latex_name="\\Delta G")
var("pCOS",latex_name="\\ p_COS")
var("pCO",latex_name="\\ p_CO")

pCO

### WATER REACTION:
#### * Eq(1) 
$$2H_2 + O_2 \rightarrow 2H_2O$$

For half of the moles on both sides, we get:
<br>$\Delta$G= -2,47,500 + 55.85*T

In [138]:
#So for doubling the moles as the Gibbs energy is extensive property.

var("delG1",latex_name="\\Delta G_1")
delG1.show()
delG1 =2*(-247500+55.85*T)
delG1.show()

delG1

111.700000000000*T - 495000

### CHROMIUM REACTION
#### * Eq(2)

$$\frac{2}{3}Cr + O_2 \rightarrow \frac{2}{3}Cr_2O_3 $$

For reaction with 1 mole of cromium and 3/2 moles and Oxygen producing one mole of $Cr_2O_3$:
<br>The given $\Delta$G = -1,110,100 + 247.3*T

In [139]:
#So as the required is 2/3 times of the original equation .So as the Gibbs energy is extensive:

var("delG2",latex_name="\\Delta G_2")
delG2.show()
delG2 =float(2/3)*(-1110100+247.3*T)
delG2.show()

delG2

164.866666666667*T - 740066.6666666666

#### Evaluating the differences between equations 2 and 1 gives us:
<br>2/3$Cr$+2$H_2O$ $\rightarrow$ 2/3$Cr_2$ + 2$H_2$

In [140]:
var('DG')
DG =(delG2-delG1)
DG.show()

53.1666666666667*T - 245066.66666666663

### Expression used to solve:
<b>Equilibrium constant</b> :
$$K= {\exp{\left(-\frac{\Delta G}{RT}\right)}}$$

Initializing variables in the above expression: R,T and doing the math

In [141]:
R=8.314
T_info = T ==1500
T_info

T == 1500

In [142]:
value = -(DG.subs(T_info) / (R*T.subs(T_info)))        
Keq=exp(value)
Keq

571538.688438923

As we know that at equilibrium there exists the maximum water pressure. 
<br>Let $P_m$ denote this maximum Pressure.
<br><br>$$K= {\frac{1}{P_m^2}}$$

In [143]:
Pm=sqrt(1/Keq)
print("Maximum Pressure obtained is",Pm,"atm.")

Maximum Pressure obtained is 0.00132274821172805 atm.


**NOTE:**
* If the pressure is above the maximum pressur ethen the $\Delta$G of equation will become negative and oxidation of Chromium will proceed.
* Thus the pressure is maximum water pressure to which Chromium can be heated without oxidizing.

Here, we get the result of change in enthalpy as: 
$\Delta H$= -245067.
(This is obtained from the Gibbs energy change obtained)

Thus the reaction is Exothermic

# Problem 2 
A gas mixture initially containing 90% CO, 0.4% COS, and 9.6% inert constituents (by volume) is passed over sponge iron at 1000 K to remove sulfur by the following reaction:

$$COS(g)+Fe(s) = CO(g)+FeS(s)$$

1. Assuming that the effluent gas is in equilibrium with Fe and FeS, calculate the percentage of sulfur removed from the gas by reaction with the sponge iron.
2. Calculate the partial pressure of $S_2$ in the effluent gas.

## 1 Solution

We need to consider three reactions in equlibrium
1. $$C(s) + \frac{1}{2}O_2(g) = CO$$
2. $$C(s) + \frac{1}{2}O_2 + \frac{1}{2}S_2 = COS$$
3. $$Fe(s) + \frac{1}{2}S_2 = FeS_2$$

These three reactions together $( 1-2+3 )$ gives us the final equlibrium reaction 
4. $$COS(g)+Fe(s) = CO(g)+FeS(s)$$

In [93]:
var('T')
CO_initial = 0.9
COS_initial = 0.004
T_info = T == 1000

In [129]:
var("delG1",latex_name="\\Delta G_1")
delG1.show()
delG1 = -111700 - 87.65*T
delG1.show()

var("delG2",latex_name="\\Delta G_2")
delG2.show()
delG2 = -202800 - 9.96*T
delG2.show()

var("delG3",latex_name="\\Delta G_3")
delG3.show()
delG3 = -150200 + 52.55*T
delG3.show()

var("delG4",latex_name="\\Delta G_4")
delG4.show()
delG4 = delG1 - delG2 + delG3
delG4.show()

delG1

-87.6500000000000*T - 111700

delG2

-9.96000000000000*T - 202800

delG3

52.5500000000000*T - 150200

delG4

-25.1400000000000*T - 59100

At 1000K, $\Delta G_r$ of the reaction can be calculated as :

In [130]:
var('delGr',latex_name="\\Delta G_r")
delGr.show()
delGr = delG4.subs(T_info)
delGr.show()

delGr

-84240.0000000000

The equlibrium constant for the reaction can be found from $\Delta G_r$ by :
$$K = exp \left({-\frac{\Delta G_r}{RT}}\right)$$

In [131]:
var("Keq1",latex_name="K_{eq}^1")
Keq1.show()
Keq1 = exp(-delGr/(R*T.subs(T_info)))
Keq1.show()

Keq1

25142.2996605887

Since Sulphur is only present in COS and there is one Sulphur atom per every COS molecule, the change in amount of sulphur is same as change in amount of COS.

With the reaction in mind: $$COS(g)+Fe(s) = CO(g)+FeS(s)$$
The following table shows the change in volume fractions of the respective gases.

|%V/V of|COS|CO|
|--|--|--|
|Initial|0.004|0.9|
|Final|0.004-x|0.9+x|

The equlibrium constant can be represented in terms of these mole fractions as :
$$K_{eq}^1 = \frac{\chi_{Co}}{\chi_{COS}}$$
Substituting volume fractions calcel out all other terms which gives us :
$$K_{eq}^1 = \frac{0.9+x}{0.004-x}$$
Equating both $K_{eq}$ will give us the value of $x$

In [132]:
var('x')
exp1 = Keq1 == (CO_initial+x)/(COS_initial-x)
exp1.show()

25142.2996605887 == -(x + 0.900000000000000)/(x - 0.00400000000000000)

solving for  $x$

In [133]:
xeq = solve(exp1,x)
COS_removed = x.subs(xeq)
print("V/V% of COS removed is given by: ")
COS_removed.n()

V/V% of COS removed is given by: 


0.00396404608733924

In [134]:
percentage_removed = COS_removed / COS_initial *100

Percentage of sulphur removed

In [100]:
percentage_removed.n()

99.1011521834811

## 2 Solution

In [85]:
var('Keq2',latex_name="K_{eq}^2")
Keq2.show()
Keq2 = exp(-delG3.subs(T_info)/(R*T.subs(T_info)))
Keq2.show()

Keq2

126152.778302548

In [88]:
var('pS',latex_name="p_{S_2}")
exp2 = K3 == 1/pS^0.5
exp2.show()
pS0 = pS.subs(solve(exp2,pS)).n()

126152.778302548 == 1/sqrt(pS)

The partial pressure of $S_2$ in the effluent gas is:

In [89]:
pS0

6.28356860114702e-11